app/src/main/java/com/example/android/view/MainActivity.java

```diff
@@ -16,34 +17,43 @@
 public class MainActivity extends AppCompatActivity {
     private RestroomsViewModel _viewModel;
     private RecyclerView _restroomRecyclerView;
+    private static final String RESTROOM_INTENT_FLAG = "restroomIntentFlag";
+    private RestroomListAdapter _adapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        RestroomListAdapter adapter = new RestroomListAdapter(this);
-        _viewModel = new RestroomsViewModel(adapter);
+        _adapter = new RestroomListAdapter(this);
+        _viewModel = new RestroomsViewModel(_adapter);
 
         _restroomRecyclerView = findViewById(R.id.recyclerView);
-        _restroomRecyclerView.setAdapter(adapter);
+        _restroomRecyclerView.setAdapter(_adapter);
         _restroomRecyclerView.setLayoutManager(new LinearLayoutManager(this));
 
-        new BasicAsyncTask(() -> _viewModel.loadRestroomsData(), () -> adapter.notifyDataSetChanged()).execute();
+        new BasicAsyncTask(() -> _viewModel.loadRestroomsData(), _adapter::notifyDataSetChanged).execute();
```

Comment:
Could you please help extract this `Runnable`?

```suggestion
        final Runnable backgroundTask = () -> _viewModel.loadRestroomsData();
        new BasicAsyncTask(backgroundTask, _adapter::notifyDataSetChanged).execute();
```